PIL/Image.py

```diff
@@ -2250,9 +2250,11 @@ def open(fp, mode="r"):
         filename = ""
 
     try:
-        fp.seek(0)
-    except (AttributeError, io.UnsupportedOperation):
-        fp = io.BytesIO(fp.read())
+        requires_iobytes_wrapper = not fp.seekable() 
+    except AttributeError:
+        # Not a subclass of io.IOBase; probably Python 2.7 file (or urllib object)
+        # Check we have a seek fn on the object
+        requires_iobytes_wrapper = not callable( getattr(fp, "seek", None) )
```

Comment:
What if there is no `seekable` attribute, but calling `seek` will raise `io.UnsupportedOperation`?